.changeset/tiny-guests-wash.md

```diff
@@ -0,0 +1,10 @@
+---
+'@xstate/vue': minor
+---
+
+chore(dep/vue) bump peerDep of @vue/composition-api to 0.6.x.
+
+- breaking changes in 0.6.x, adapated code. Will make it easier for
+  this library to also be updated to Vue 3.
+- Uses new watcher option since it was changed to be lazy by default
+- now using shallow ref since xstate objects should not be refs deeply.
```

Comment:
Prefixes like `chore(dep/vue)` are not needed, the changelog entry created later because of this changeset file will only appear in the `./packages/xstate-vue/CHANGELOG.md` so no need to include this information here. A changeset should also only describe user-facing changes, no need to mention that, for example, a new watcher option has been used internally - it's just an implementation detail.

I propose to change this to something like this:
```suggestion
Upgraded package for compatibility with the newest `@vue/composition-api@^0.6.0`, which means that a peer dependency requirement has changed to this version, which is a **breaking change**. The only observable behavior change is that exposed refs are now **shallow**.
```

Which assumes that previously exposed refs were deep, which I don't know if it was true nor what does it actually mean in context of Vue. Would love to learn more if you have a spare moment to explain this.

Reply:
Migrating to 0.6 _seems_ to have changed how `ref` returns were typed. So actually I think it _could_ work with traditional ref's if the type error could be resolved:

```sh
Type 'State<TContext, TEvent, TState>' is not assignable to type '{ value: string; context: UnwrapRef<TContext>; actions: ActionObject<TContext, TEvent>[]; changed?: boolean | undefined; matches: <TSV extends TState["value"]>(value: TSV) => this is TState extends { ...; } ? TState : never; }'.
  Types of property 'context' are incompatible.
    Type 'TContext' is not assignable to type 'UnwrapRef<TContext>'.
```
^^not sure how to resolve the `UnwrapRef<TContext>` typing.

A traditional ref will allow for an object's properties and sub-properties to be reactive. The "gotcha" with a shallowRef is that members are not reactive. `shallowRef`'s continue to allow `@xstate/vue` to work since all `ref` assignments used are wholesale replaced e.g. `current.value = ` or `state.value = ` (vs. `state.value.value =`). However, this is a breaking change if users were wanting to mutate return ref's properties directly and still have reactivity like today, e.g. if `state.value.value` is assigned directly instead of via `send`, it would not trigger any vue reactivity (watchers, computed, template updates etc.)

Reply:
> A traditional ref will allow for an object's properties and sub-properties to be reactive.

Interesting, is it Proxy-based?

Reply:
the composition-api is not proxy based, though `vue-next` (v3) is